vmdk_plugin/drivers/shared/shared_driver.go

```diff
@@ -0,0 +1,209 @@
+	log.Errorf("VolumeDriver UnmountVolume to be implemented")
+	return nil
+}
+
+// Create - create a volume.
+func (d *VolumeDriver) Create(r volume.Request) volume.Response {
+	log.Errorf("VolumeDriver Create to be implemented")
+	return volume.Response{Err: ""}
+}
+
+// Remove - removes individual volume. Docker would call it only if is not using it anymore
+func (d *VolumeDriver) Remove(r volume.Request) volume.Response {
+	log.WithFields(log.Fields{"name": r.Name}).Info("Removing volume ")
+
+	// Cannot remove volumes till plugin completely initializes (refcounting is complete)
+	// because we don't know if it is being used or not
+	if d.refCounts.IsInitialized() != true {
+		msg := fmt.Sprintf(plugin_utils.PluginInitError+" Cannot remove volume=%s", r.Name)
+		log.Error(msg)
+		return volume.Response{Err: msg}
+	}
+
+	// Docker is supposed to block 'remove' command if the volume is used.
+	if d.getRefCount(r.Name) != 0 {
+		msg := fmt.Sprintf("Remove failure - volume is still mounted. "+
+			" volume=%s, refcount=%d", r.Name, d.getRefCount(r.Name))
+		log.Error(msg)
+		return volume.Response{Err: msg}
+	}
+
+	log.Errorf("VolumeDriver Remove to be implemented")
+	return volume.Response{Err: ""}
+}
+
+// Path - give docker a reminder of the volume mount path
+func (d *VolumeDriver) Path(r volume.Request) volume.Response {
+	return volume.Response{Mountpoint: getMountPoint(r.Name)}
+}
+
+// Mount - Provide a volume to docker container - called once per container start.
+// We need to keep refcount and unmount on refcount drop to 0
+//
+// The serialization of operations per volume is assured by the volume/store
+// of the docker daemon.
+// As long as the refCountsMap is protected is unnecessary to do any locking
+// at this level during create/mount/umount/remove.
+//
+func (d *VolumeDriver) Mount(r volume.MountRequest) volume.Response {
+	log.WithFields(log.Fields{"name": r.Name}).Info("Mounting volume ")
+
+	// lock the state
+	d.refCounts.StateMtx.Lock()
+	defer d.refCounts.StateMtx.Unlock()
+
+	log.Errorf("VolumeDriver Mount to be implemented")
+	return volume.Response{Err: ""}
+}
+
+// Unmount request from Docker. If mount refcount is drop to 0.
+func (d *VolumeDriver) Unmount(r volume.UnmountRequest) volume.Response {
+	log.WithFields(log.Fields{"name": r.Name}).Info("Unmounting Volume ")
+
+	// lock the state
+	d.refCounts.StateMtx.Lock()
+	defer d.refCounts.StateMtx.Unlock()
+
+	if d.refCounts.IsInitialized() != true {
+		// if refcounting hasn't been succesful,
+		// no refcounting, no unmount. All unmounts are delayed
+		// until we succesfully populate the refcount map
+		d.refCounts.MarkDirty()
+		return volume.Response{Err: ""}
+	}
+
+	log.Errorf("VolumeDriver Unmount to be implemented")
+	return volume.Response{Err: ""}
+}
+
+// Capabilities - Report plugin scope to Docker
+func (d *VolumeDriver) Capabilities(r volume.Request) volume.Response {
+	return volume.Response{Capabilities: volume.Capability{Scope: "global"}}
```

Comment:
SO I guess the text above will change for actual shared- driver and is cut-n-paste just do the split, right ?